### PR TITLE
Minor Wires fixes

### PR DESCRIPTION
**Pull request by nicolatimeus**
Fixed possible NPE in `GwtWiresServiceImpl`
`OpcUaDriver.getChannelDescriptor()` is no longer synchronized

Signed-off-by: nicolatimeus <nicola.timeus@eurotech.com>

### CODE REVIEW

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java [hunk 1]
```diff
@@ -530,23 +529,23 @@ private void removeDeletedFromWireGraphProperty(final String pid) throws GwtKura
...
             if (props != null) {
                 props.remove(DELETED_WIRE_COMPONENT);
-            }
-            if (conf != null) {
                 conf.update(props);
             }
-        } catch (final IOException e) {
-            // no need
+        } catch (final Exception exception) {
+            throw new GwtKuraException(GwtKuraErrorCode.INTERNAL_ERROR, exception);
```
Comment (amitjoy) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:550]: Use some specific Error Code

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java [hunk 2]
```diff
@@ -530,23 +529,23 @@ private void removeDeletedFromWireGraphProperty(final String pid) throws GwtKura
         final ConfigurationAdmin configAdmin = ServiceLocator.getInstance().getService(ConfigurationAdmin.class);
         final WireHelperService wireHelperService = ServiceLocator.getInstance().getService(WireHelperService.class);
         try {
+            final String filter = "(" + KURA_SERVICE_PID + "=" + pid + ")";
+            ServiceUtil.waitForService(filter, TIMEOUT, SECONDS);
```
Comment (amitjoy) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java]: Check if the returned optional is empty and if so return.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java (hunk 1 above)
Comment (amitjoy) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:550]: As INTERNAL_ERROR is deprecated, it is not advised.

Comment (nicolatimeus) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:550]: I wasn't able to find any better enum constant for this case. Let's postpose this

Comment (amitjoy) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:550]: You can catch three exceptions in two catch blocks - `IOException, InvalidSyntaxException` and `InterruptedException`

`IOException, InvalidSyntaxException`  -> `CONFIGURATION_ERROR` (as it is related to ConfigAdmin and filter)
`InterruptedException` -> `PROCESS_EXECUTION_ERROR`

Comment (nicolatimeus) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:550]: It is` GwtKuraErrorCode` not `KuraErrorCode`, the available enum variants are limited.

**Review by amitjoy — APPROVED** [review]

**Pull request merged**